File: iosApp/iosApp/Presentation/Main/CafeManagement/CafeManagementAction.swift
//
//  CafeManagementAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/07.
//

import Foundation

enum CafeManagementAction {
    case clickCafe(String)
    case clickCafeDetail(String)
    case changeCafeSearchQuery(String)
    case clickClaimCafe(String)
    case toggleCafeListExpanded
    case clickCreateCafe
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/CafeManagementAction.kt
package com.hhp227.concafe.presentation.main.cafemanagement

sealed interface CafeManagementAction {
    data class ClickCafe(val cafeId: String) : CafeManagementAction
    data class ClickCafeDetail(val cafeId: String) : CafeManagementAction
    data class ChangeCafeSearchQuery(val query: String) : CafeManagementAction
    data class ClickClaimCafe(val cafeId: String) : CafeManagementAction
    data object ToggleCafeListExpanded : CafeManagementAction
    data object ClickCreateCafe : CafeManagementAction
    data object DismissInfoMessage : CafeManagementAction
}
